core/chains/evm/txmgr/finalizer.go

```diff
@@ -666,3 +675,64 @@ func (f *evmFinalizer) buildTxHashList(finalizedReceipts []*types.Receipt) []com
 	}
 	return txHashes
 }
+
+// fetchAttemptsRequiringReceiptFetch is a wrapper around the TxStore call to fetch attempts requiring receipt fetch.
+// Attempts are cached and used for subsequent fetches to reduce the load of the query.
+// The attempts cache is refreshed every 3 requests.
+func (f *evmFinalizer) fetchAttemptsRequiringReceiptFetch(ctx context.Context) ([]TxAttempt, error) {
+	f.attemptsCacheMu.Lock()
```

Comment:
Is locking necessary?

Reply:
Probably not since the two calls that use the cache are sequential. I added it just in case there's any changes to it in the future since it's not obvious that this needs to be protected if it's called in a thread unsafe way. I'm probably getting ahead of myself though so I can remove it